### PR TITLE
script and playbook to increase number of worker nodes

### PR DESCRIPTION
**Pull request by andywis**
This is the Ansible side. There's also a change in the Heat repo

### CODE REVIEW

File: tools/create-upscale-inventory.py [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+
+    for host, ip_arg in new_hosts:
+        # the new host is either "master-*.openstacklocal" or
+        # worker-*.openstacklocal"
+
+        host_type = ''
+        host_num = -1
+        if debug:
+            print("DEBG: {}".format(host))
+        mtch = re.match('(worker|master)-(\d+).openstacklocal', host)
```
Comment (andywis) [inline tools/create-upscale-inventory.py]: it's not always openstacklocal - we could read the precise hostname from the config

**Comment by andywis** [comment]
Following review, I've reorganised the python a bit, and removed references to "openstacklocal" in the code (they're still present in the tests, which is fine, AFAIK)

**Comment by andywis** [comment]
Added a test that builds an app and runs it on each node, to ensure the upscale was successful.

**Review by stevemul — COMMENTED** [review]
This looks good to me. I think my main comment is actually that the upscale tests would be useful before having upscaled. We should be running those on the workers that got deployed before we upscaled too - so maybe we can generalise the names and just call it workers tests?

**Review by stevemul — APPROVED** [review]
lgtm

**Pull request merged**